base/loading.jl

```diff
@@ -295,16 +297,17 @@ then tries paths in the global array `LOAD_PATH`. `require` is case-sensitive on
 all platforms, including those with case-insensitive filesystems like macOS and
 Windows.
 """
-function require(mod::Symbol)
+function require(from::Module, mod::Symbol)
     if !root_module_exists(mod)
-        _require(mod)
+        info("@$(getpid()): require($from, $mod)")
+        _require(from, mod)
         # After successfully loading, notify downstream consumers
         if toplevel_load[] && myid() == 1 && nprocs() > 1
```

Comment:
This codepath is moving out of base with #25139, so the `callback` below will need to also be changed to include `from`.